### PR TITLE
feat(kinesis): support for enabling enhanced stream monitoring

### PR DESCRIPTION
**Pull request by pedrinhok**
Since CloudFormation doesn't support creating a stream with enhanced monitoring enabled (there is an [open issue](https://github.com/aws-cloudformation/cloudformation-coverage-roadmap/issues/768) on their side), we leverage StepFunctions to call Kinesis API and enable enhanced monitoring of an existing stream.

----

### All Submissions:

* [X] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [X] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [X] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by Naumel** [comment]
Hi, based on the PR description alone, I have some questions:
> we leverage StepFunctions to call Kinesis API and enable enhanced monitoring of an existing stream.

1. Why is this design the best option?
2. What other choices were considered?

The change is not trivial, a bit more information around it is always welcome.

Additionally, from the liked feature request, there's an update from June 13th:

> @[cfn-github-issues-bot](https://github.com/cfn-github-issues-bot) cfn-github-issues-bot moved this from We're working on it to Coming Soon in [coverage-roadmap](https://github.com/aws-cloudformation/cloudformation-coverage-roadmap/projects/1#card-67109782) [on 13 Jun](https://github.com/aws-cloudformation/cloudformation-coverage-roadmap/issues/768#event-6798876381)

Is this solely a work-around for a missing feature in CFn or adding intrinsic value to CDK?

**Comment by pedrinhok** [comment]
Hi Namuel,

Answering your questions:

> we leverage StepFunctions to call Kinesis API and enable enhanced monitoring of an existing stream.
> 1. Why is this design the best option?
> 2. What other choices were considered?

This is currently the only choice we have, as there is no support for enabling enhanced stream monitoring if not by calling [EnableEnhancedMonitoring action](https://docs.aws.amazon.com/kinesis/latest/APIReference/API_EnableEnhancedMonitoring.html) from Kinesis API.

> Additionally, from the liked feature request, there's an update from June 13th:

If you check the [issue updates](https://github.com/aws-cloudformation/cloudformation-coverage-roadmap/issues/768) you will see it has been moving between "We're working on it" and "Coming Soon" over 1 year now, so it's hard to know when exactly we will have this feature in CFn (if ever).

> Is this solely a work-around for a missing feature in CFn or adding intrinsic value to CDK?

I would say both. There is the lack of support from CFn to enable enhanced monitoring while creating a stream, but there is also the benefit of providing this ability in CDK to enable enhanced monitoring of an existing stream.

**Comment by corymhall** [comment]
> we leverage StepFunctions to call Kinesis API 

How is the step function triggered? Does it have to be triggered manually after the deployment? Typically we handle this type of thing using a custom resource.

File: packages/@aws-cdk/aws-stepfunctions-tasks/test/kinesis/integ.enable-enhanced-monitoring.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  testCases: [stack],
+});
+
+// start the execution
+const start = testCase.assertions.awsApiCall('StepFunctions', 'startExecution', {
+  stateMachineArn: sm.stateMachineArn,
+});
+
+// describe the results of the execution
+const describe = testCase.assertions.awsApiCall('StepFunctions', 'describeExecution', {
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-stepfunctions-tasks/test/kinesis/integ.enable-enhanced-monitoring.ts:45]: this feature is meant to be used via API call? If we're going to do this, it should include a custom resource, like @corymhall said.

**Comment by pedrinhok** [comment]
Okay, apparently I was just misinformed (my bad), we can already enable enhanced stream monitoring by making use of custom resources. Tbh I was totally unaware of the existence (and power) of this feature.

Thanks @corymhall and @comcalvi for pointing that out!

Anyway, cancelling the PR as we can obtain the same result with custom resources (see the [docs](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.custom_resources.AwsCustomResource.html)).

**Pull request closed without merging**